proposals/csharp-9.0/unconstrained-type-parameter-annotations.md

```diff
@@ -0,0 +1,67 @@
+﻿# Unconstrained type parameter annotations
+
+## Summary
+
+Allow nullable annotations for type parameters that are not constrained to value types or reference types: `T?`.
+```C#
+static T? FirstOrDefault<T>(this IEnumerable<T> collection) { ... }
+```
+
+## `?` annotation
+
+In C#8, `?` annotations could only be applied to type parameters that were explicitly constrained to value types or reference types.
+In C#9, `?` annotations can be applied to any type parameter, regardless of constraints.
+
+If a type parameter `T` is substituted with a reference type, then `T?` represents a nullable instance of that reference type.
+If `T` is substituted with a value type, then `T?` is represents an instance of `T`.
+```C#
+var s1 = new string[0].FirstOrDefault();  // string? s1
+var s2 = new string?[0].FirstOrDefault(); // string? s2
+var i1 = new int[0].FirstOrDefault();     // int i1
+var i2 = new int?[0].FirstOrDefault();    // int? i2
```

Comment:
One case not covered in the samples or text is what happens when `T` is substituted with a type parameter `U` which is not definitely constrained to a reference or value type. Essentially 

``` 
T? M1<T>() => throw null!;
void M2<U>()
{
  var local1 = M1<U>();   // what is the type here?
  var local2 = M1<U?>(); // what is the type here?
}
```